api/ws/rpc/connect.js

```diff
@@ -199,8 +199,14 @@ const connectSteps = {
 					peer.ip
 				} failed with code ${code} and reason - ${reason}`
 			);
-			socket.destroy();
-			onDisconnectCb();
+
+			if (peer.socket && peer.socket.state === peer.socket.CLOSED) {
+				peer.state = Peer.STATE.DISCONNECTED;
```

Comment:
why are we changing the state from `closed` to `disconnected`?

Reply:
@ManuGowda In one case, we're comparing the socket state and in the other, we're comparing the peer state.